### DISCUSSION TITLE
Regarding metamask integration

### DISCUSSION THREAD

**Discussion by anithakc6** [gochain/web3#245] [General]
Can I deploy and write date using the commans, connecting to metamask and giving signer to these commands?
The reason is I don't want to share the private key. 
Thanks in advance

**Comment by treeder** [comment]
No, this won't work with metamask, metamask requires a web app.

**Reply by anithakc6** [reply]
Thanks for the quick reply, so storing the private key is the only way we have to proceed doing write operations? but its not secured way right?

**Reply by treeder** [reply]
It's as secure as you make it. If you're the only one accessing the command line on that computer, it should be fine. If it's a shared computer, I wouldn't do it. But also depends what you're doing and how secure that key needs to be. Metamask is used so websites don't steal your key, the web3 cli isn't a website you don't trust, so it's a much different use case.

**Reply by anithakc6** [reply]
Great, thanks for the answers, appreciate your time
